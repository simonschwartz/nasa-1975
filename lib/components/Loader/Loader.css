.NASA1975-loader-container {
  display: inline-block;
  text-align: center;
  animation-name: NASA1975-loading__aniamtion__fadeIn;
  animation-duration: 1s;
  animation-timing-function: linear;
  animation-delay: .25s;
  animation-iteration-count: 1;
  animation-fill-mode: both;
}

.NASA1975-loader-outer {
  position: relative;
  margin: auto;
}

.NASA1975-loader-container--lg .NASA1975-loader-outer {
  width: 60px;
  height: 60px;
}

.NASA1975-loader-container--md .NASA1975-loader-outer {
  width: 40px;
  height: 40px;
}

.NASA1975-loader-container--sm .NASA1975-loader-outer {
  width: 20px;
  height: 20px;
}

.NASA1975-loader-container--lg .NASA1975-loader-inner {
  border-top: 6px solid #f4f4f4;
  border-right: 6px solid #f4f4f4;
  border-bottom: 6px solid #f4f4f4;
  border-left: 6px solid #e30613;
}

.NASA1975-loader-container--md .NASA1975-loader-inner {
  border-top: 4px solid #f4f4f4;
  border-right: 4px solid #f4f4f4;
  border-bottom: 4px solid #f4f4f4;
  border-left: 4px solid #e30613;
}

.NASA1975-loader-container--sm .NASA1975-loader-inner {
  border-top: 3px solid #f4f4f4;
  border-right: 3px solid #f4f4f4;
  border-bottom: 3px solid #f4f4f4;
  border-left: 3px solid #e30613;
}

.NASA1975-loader-inner {
  font-size: 6px;
  position: relative;
  border-top: 4px solid #f4f4f4;
  border-right: 4px solid #f4f4f4;
  border-bottom: 4px solid #f4f4f4;
  border-left: 4px solid #e30613;
  border-radius: 50%;
  width: 100%;
  height: 100%;
  animation-name: NASA1975-loading__aniamtion__spin;
  animation-duration: 1s;
  animation-timing-function: linear;
  animation-delay: .25s;
  animation-iteration-count: infinite;
}

.NASA1975-loader-message {
  display: block;
  color: #4e4e4e;
  font-size: 15px;
  margin-top: 6px;
}

.NASA1975-loader-container--inverted .NASA1975-loader-message {
  color: #fff;
}

@keyframes NASA1975-loading__aniamtion__spin {
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
}

@keyframes NASA1975-loading__aniamtion__fadeIn {
  0% {
    opacity: 0;
  }

  100% {
    opacity: 1;
  }
}
