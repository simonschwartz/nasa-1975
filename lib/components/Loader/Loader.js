import React from "react";
import PropTypes from "prop-types";
import "./Loader.css";

const Loader = ({
  size = "md",
  className = "",
  message = "",
  inverted = false,
  ...attributeOptions
}) => {
  const invertedClassName = inverted
    ? "NASA1975-loader-container--inverted"
    : "";
  return (
    <div
      className={`NASA1975-loader-container NASA1975-loader-container--${size} ${invertedClassName} ${className}`}
      {...attributeOptions}
    >
      <div className={`NASA1975-loader-outer`}>
        <div className={`NASA1975-loader-inner`} />
      </div>
      {message &&
        <span className={`NASA1975-loader-message`}>
          {message}
        </span>}
    </div>
  );
};

Loader.propTypes = {
  /** Size of the loader */
  size: PropTypes.oneOf(["sm", "md", "lg"]),
  /** Optional text to display below loader */
  message: PropTypes.string,
  /** Invert the loader colours when used on dark background */
  inverted: PropTypes.bool,
  /** className to add to loader container element */
  className: PropTypes.string
};

export default Loader;
