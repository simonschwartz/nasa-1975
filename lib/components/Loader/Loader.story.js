import React from "react";
import { text, select, boolean } from "@storybook/addon-knobs";
import { withInfo } from "@storybook/addon-info";
import Loader from "./";

const LoaderStory = withInfo({
  text:
    "Loading icons weren't around in 1975, but its really hard to build application UIs without a loading icon. This is a simple loader that users the NASA 1975 colour scheme."
})(() => {
  // Setup knob for setting loader size
  const labelSize = "size";
  const optionsSize = {
    lg: "lg",
    md: "md",
    sm: "sm"
  };
  const defaultValueSize = "md";
  const sizeValue = select(labelSize, optionsSize, defaultValueSize);

  // Setup knob for setting loader message
  const labelMessage = "message";
  const defaultValueMessage = "-loading-";
  const messageValue = text(labelMessage, defaultValueMessage);

  // Setup knob for setting inverted state
  const labelInverted = "inverted";
  const defaultValueInverted = false;
  const invertedValue = boolean(labelInverted, defaultValueInverted);

  return (
    <Loader size={sizeValue} message={messageValue} inverted={invertedValue} />
  );
});

export default LoaderStory;
