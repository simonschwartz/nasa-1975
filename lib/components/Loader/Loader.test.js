import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Loader from "./";

it("renders correct markup with no props", () => {
  const tree = renderer.create(<Loader />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correct markup with message prop", () => {
  const tree = renderer.create(<Loader message="-loading message-" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correct markup with inverted prop", () => {
  const tree = renderer.create(<Loader inverted />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correct markup for lg size prop", () => {
  const tree = renderer.create(<Loader size="lg" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correct markup for md size prop", () => {
  const tree = renderer.create(<Loader size="md" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correct markup for sm size prop", () => {
  const tree = renderer.create(<Loader size="sm" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Loader />, div);
});
