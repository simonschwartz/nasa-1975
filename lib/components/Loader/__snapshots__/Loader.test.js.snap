// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct markup for lg size prop 1`] = `
<div
  className="NASA1975-loader-container NASA1975-loader-container--lg  "
>
  <div
    className="NASA1975-loader-outer"
  >
    <div
      className="NASA1975-loader-inner"
    />
  </div>
  
</div>
`;

exports[`renders correct markup for md size prop 1`] = `
<div
  className="NASA1975-loader-container NASA1975-loader-container--md  "
>
  <div
    className="NASA1975-loader-outer"
  >
    <div
      className="NASA1975-loader-inner"
    />
  </div>
  
</div>
`;

exports[`renders correct markup for sm size prop 1`] = `
<div
  className="NASA1975-loader-container NASA1975-loader-container--sm  "
>
  <div
    className="NASA1975-loader-outer"
  >
    <div
      className="NASA1975-loader-inner"
    />
  </div>
  
</div>
`;

exports[`renders correct markup with inverted prop 1`] = `
<div
  className="NASA1975-loader-container NASA1975-loader-container--md NASA1975-loader-container--inverted "
>
  <div
    className="NASA1975-loader-outer"
  >
    <div
      className="NASA1975-loader-inner"
    />
  </div>
  
</div>
`;

exports[`renders correct markup with message prop 1`] = `
<div
  className="NASA1975-loader-container NASA1975-loader-container--md  "
>
  <div
    className="NASA1975-loader-outer"
  >
    <div
      className="NASA1975-loader-inner"
    />
  </div>
  <span
    className="NASA1975-loader-message"
  >
    -loading message-
  </span>
</div>
`;

exports[`renders correct markup with no props 1`] = `
<div
  className="NASA1975-loader-container NASA1975-loader-container--md  "
>
  <div
    className="NASA1975-loader-outer"
  >
    <div
      className="NASA1975-loader-inner"
    />
  </div>
  
</div>
`;
